Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt

```diff
@@ -9,46 +9,47 @@ import androidx.fragment.app.Fragment
 import de.rki.coronawarnapp.R
 import de.rki.coronawarnapp.databinding.FragmentOnboardingBinding
 import de.rki.coronawarnapp.util.di.AutoInject
-import de.rki.coronawarnapp.util.ui.observe2
-import de.rki.coronawarnapp.util.ui.viewBindingLazy
-import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
-import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
 import de.rki.coronawarnapp.util.ui.doNavigate
-import javax.inject.Inject
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import java.util.Locale
 
 /**
  * Onboarding starting point.
  */
 class OnboardingFragment : Fragment(R.layout.fragment_onboarding), AutoInject {
 
-    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
-    private val vm: OnboardingFragmentViewModel by cwaViewModels { viewModelFactory }
     private val binding: FragmentOnboardingBinding by viewBindingLazy()
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding.apply {
-            onboardingButtonNext.setOnClickListener { vm.onNextButtonClick() }
-            onboardingInclude.onboardingEasyLanguage.setOnClickListener { vm.onEasyLanguageClick() }
-        }
-        vm.routeToScreen.observe2(this) {
-            when (it) {
-                is OnboardingNavigationEvents.NavigateToOnboardingPrivacy ->
-                    doNavigate(
-                        OnboardingFragmentDirections
-                            .actionOnboardingFragmentToOnboardingPrivacyFragment()
-                    )
-                is OnboardingNavigationEvents.NavigateToEasyLanguageUrl ->
-                    startActivity(
-                        Intent(
-                            Intent.ACTION_VIEW,
-                            Uri.parse(getString(R.string.onboarding_tracing_easy_language_explanation_url))
-                        )
-                    )
+            onboardingButtonNext.setOnClickListener { goToOnboardingPrivacyFragment() }
+            // only show link for German
+            if (Locale.getDefault().language == "de") {
```

Comment:
This check could be implemented in the VM instead. The VM could expose a boolean `onboardingEasyLanguageEnabled` and a test could be written by easily mocking the Locale.

Reply:
I will move to a function. This way I can write a unit test and don't need a view model.